lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -15,12 +15,40 @@ class FirebaseAppDistributionAction < Action
       FIREBASECMD_ACTION = "appdistribution:distribute".freeze
       BASE_URL = "https://firebaseappdistribution.googleapis.com"
       PATH = "/v1alpha/apps/"
+      MAX_POLLING_RETRIES = 60
+      POLLING_INTERVAL_S = 2
+      CONNECTION = Faraday.new(url: BASE_URL) do |conn|
+        conn.response(:json, parser_options: { symbolize_names: true })
+        conn.response(:raise_error) # raise_error middleware will run before the json middleware
+        conn.adapter(Faraday.default_adapter)
+      end
 
       extend Helper::FirebaseAppDistributionHelper
 
       def self.run(params)
         params.values # to validate all inputs before looking for the ipa/apk
-        validate_app!(params[:app])
+        token_string = validate_app!(params[:app])
+        begin
+          for checks in 1..MAX_POLLING_RETRIES
+            status = upload_status!(token_string, params[:app], params[:apk_path])
+            if status == "SUCCESS"
+              UI.message("App Uploaded Successfully!") # Probably want something like it had before of apk has already been uploaded so no need to upload again
+              break
+            elsif status == "IN_PROGRESS"
+              UI.message("Still in progress...")
```

Comment:
Will this print out after each poll? I think we only need to print this out a status once before we upload i.e `Uploading APK to Firebase App Distribution...`